pkg/asset/agent/mirror/registriesconf.go

```diff
@@ -187,7 +210,25 @@ func (i *RegistriesConf) Load(f asset.FileFetcher) (bool, error) {
 		return false, errors.Wrap(err, fmt.Sprintf("failed to load %s file", RegistriesConfFilename))
 	}
 
-	i.File = file
+	registriesConf := &sysregistriesv2.V2RegistriesConf{}
+	if err := toml.Unmarshal(file.Data, registriesConf); err != nil {
+		return false, errors.Wrapf(err, "failed to unmarshal %s", RegistriesConfFilename)
+	}
+
+	i.File, i.Config = file, registriesConf
+
+	mirrorConfig := make([]RegistriesConfig, len(registriesConf.Registries))
+	for i, reg := range registriesConf.Registries {
+		mirrorConfig[i] = RegistriesConfig{
+			Location: reg.Location,
+			Mirror:   reg.Mirrors[0].Location,
+		}
+	}
+	if err != nil {
+		return false, errors.Wrap(err, fmt.Sprintf("failed to parse mirrors in %s", RegistriesConfFilename))
+	}
+
+	i.MirrorConfig = mirrorConfig
```

Comment:
@andfasano When I did the restructuring, my thought was we only do the unmarshalling in the `Load()` and hence when I refactored I moved the unmarshalling code into `Load()`. Originally here https://github.com/openshift/installer/blame/master/pkg/asset/agent/mirror/registriesconf.go#L199-L209 the unmarshalling is done in `finish()` which gets called in case of `Generate()` as well as `Load()`. Looking at it now makes me think why we even need to set `MirrorConfig`. I don't see it us being used anywhere else in the code. I think the public field `MirrorConfig` should be removed completely. @bfournie WDYT?

Reply:
The MirrorConfig is used when the registries.conf is read in ignition, e.g. https://github.com/openshift/installer/blob/master/pkg/asset/agent/image/ignition.go#L150. 

It should be set in the finish() for both Load and Generate (see line 201 and 206 below).  I thinks its a good catch that its no longer being set in the Generate.

Reply:
Done